src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,111 @@
+package duke.parser;
+
+import java.util.regex.Pattern;
+
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.EventCommand;
+import duke.commands.FindCommand;
+import duke.commands.IncorrectCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.TodoCommand;
+import duke.commands.UnmarkCommand;
+import duke.enums.CommandType;
+import duke.exceptions.DukeException;
+
+
+/**
+ * Parser class to parse input command.
+ */
+public class Parser {
+
+    /**
+     * Parses an input command.
+     * @param fullCommand Input command to be parsed.
+     * @return One of the specified Command objects.
+     * @throws DukeException If command is invalid.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] arr = fullCommand.split(" ", 2);
+        String commandType = arr[0].toLowerCase();
+        if (commandType.equals(CommandType.BYE.toString())) {
```

Comment:
You can consider using a `switch` block for this section 😄

Reply:
I tried using switch but I encountered some issues using switch with enum because the switch cases need to be constants but my enum toString method somehow does not evaluate to a constant at compile time.